temporalio/testing/_activity.py

```diff
@@ -74,15 +74,27 @@ def __init__(self) -> None:
         self._cancelled = False
         self._worker_shutdown = False
         self._activities: Set[_Activity] = set()
+        self._cancellation_details = (
+            temporalio.activity._ActivityCancellationDetailsHolder()
+        )
 
-    def cancel(self) -> None:
+    def cancel(
+        self,
+        cancellation_details: temporalio.activity.ActivityCancellationDetails,
```

Comment:
To clarify, _providing_ the parameter still needs to be optional (this is a backwards incompatible change otherwise), it just needs to not be `Optional`. So:

```suggestion
        cancellation_details: temporalio.activity.ActivityCancellationDetails = temporalio.activity.ActivityCancellationDetails(cancel_requested=True),
```

I just picked `cancel_requested` as the default (might as well set one to `True` by default). Usually having an expression as a parameter default like this in Python is bad because that same instance is reused for _every_ default so mutations are shared, but this object is immutable so it's safe.

Sorry for the confusion